File: ruby/pe003.rb
#!/usr/bin/env ruby
# http://projecteuler.net/problem=3
n = 600_851_475_143
n /= 2 while n % 2 == 0

factor = 3
while n != 1
  n /= factor while n % factor == 0
  factor += 2
end
factor -= 2

puts factor

File: go/pe003.go
// http://projecteuler.net/problem=3
package main

import "fmt"

func main() {
	var number int64 = 600851475143
	var factor int64 = 1
	for number != 1 {
		factor++
		for number%factor == 0 {
			number /= factor
		}
	}
	fmt.Println(factor)
}
